pkg/agent/cniserver/pod_configuration_linux_test.go

```diff
@@ -0,0 +1,514 @@
+		},
+		{
+			name:            "error-migrate-route",
+			migratedRoute:   true,
+			migrateRouteErr: fmt.Errorf("unable to get host interface"),
+			expectedErr:     true,
+		},
+		{
+			name:             "error-ovs-create-port",
+			migratedRoute:    true,
+			connectedOVS:     true,
+			createOVSPortErr: ovsconfig.NewTransactionError(fmt.Errorf("unable to create OVS port"), true),
+			expectedErr:      true,
+		},
+		{
+			name:          "error-ovs-get-ofport",
+			migratedRoute: true,
+			connectedOVS:  true,
+			getOFPortErr:  ovsconfig.NewTransactionError(fmt.Errorf("timeout to get OpenFlow port"), true),
+			expectedErr:   true,
+		},
+		{
+			name:              "error-ovs-install-flows",
+			migratedRoute:     true,
+			connectedOVS:      true,
+			installPodFlowErr: fmt.Errorf("failed to install Pod OpenFlow"),
+			expectedErr:       true,
+		},
+		{
+			name:          "success",
+			migratedRoute: true,
+			connectedOVS:  true,
+			expectedErr:   false,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			podConfigurator := createPodConfigurator(controller, createIfaceCreator(tc.getInterfaceErr))
+			getNSPath = func(netnsName string) (string, error) {
+				return netnsName, tc.getNSPathErr
+			}
+			if tc.migratedRoute {
+				routeMock.EXPECT().MigrateRoutesToGw(hostInterfaceName).Return(tc.migrateRouteErr)
+			}
+			ovsPortID := generateUUID(t)
+			if tc.connectedOVS {
+				mockOVSBridgeClient.EXPECT().CreatePort(hostInterfaceName, gomock.Any(), gomock.Any()).Return(ovsPortID, tc.createOVSPortErr).Times(1)
+				if tc.createOVSPortErr == nil {
+					mockOVSBridgeClient.EXPECT().GetOFPort(hostInterfaceName, false).Return(int32(100), tc.getOFPortErr).Times(1)
+					if tc.getOFPortErr == nil {
+						mockOFClient.EXPECT().InstallPodFlows(hostInterfaceName, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(tc.installPodFlowErr).Times(1)
+					}
+					if tc.getOFPortErr != nil || tc.installPodFlowErr != nil {
+						mockOVSBridgeClient.EXPECT().DeletePort(ovsPortID).Times(1)
+					}
+				}
+			}
+			err := podConfigurator.connectInterceptedInterface(podName, podNamespace, containerID, containerNetNS, containerDev, ipamResult.IPs, nil)
+			if tc.expectedErr {
+				assert.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				containerConfig, exists := ifaceStore.GetContainerInterface(containerID)
+				assert.True(t, exists)
+				assert.Equal(t, containerID, containerConfig.ContainerID)
+				assert.Equal(t, ovsPortID, containerConfig.OVSPortConfig.PortUUID)
+			}
+		})
+	}
+}
+
+func TestCreateOVSPort(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	containerID := generateUUID(t)
+	podName := "p0"
+	podNameSpace := testPodNamespace
+
+	for _, tc := range []struct {
+		name                  string
+		portName              string
+		portType              int
+		vlanID                uint16
+		createPortCount       int
+		createAccessPortCount int
```

Comment:
could a single `createOVSPort` call create multiple ports? If not, it makes more sense to use booleans?

Reply:
updated as `createOVSPort` and `createOVSAccessPort`, and change the type as boolean.